chart/kubeapps/templates/kubeapps-frontend-config.yaml

```diff
@@ -61,10 +61,29 @@ data:
       location ~* /api/clusters/{{ .name }} {
         rewrite /api/clusters/{{ .name }}/(.*) /$1 break;
         rewrite /api/clusters/{{ .name }} / break;
+
+      # Helm returns a nil pointer error when accessing foo.bar if foo doesn't
+      # exist, even with the `default` function.
+      # See https://github.com/helm/helm/issues/8026#issuecomment-756538254
+      {{- $pinnipedConfig := .pinnipedConfig | default dict }}
+      {{- if and $.Values.pinnipedProxy.enabled $pinnipedConfig.exchangeCredentials }}
+        # If pinniped proxy is enabled *and* the current cluster is configured
+        # to exchange credentials then we route via pinnipedProxy to exchange
+        # credentials for client certificates.
+        {{- if .apiServiceUrl }}
+        proxy_set_header PINNIPED_PROXY_API_SERVER_URL {{ .apiServiceURL }};
+        {{- end }}
+        {{- if .certificateAuthorityData }}
+        proxy_set_header PINNIPED_PROXY_API_SERVER_CERT {{ .certificateAuthorityData }};
+        {{- end }}
+        proxy_pass http://localhost:3333;
```

Comment:
Shouldn't we also parametrize this? I believe we're doing so in other `proxy_pass` but I'm unsure if it makes sense here.

Reply:
Yes, you're right. And this would be another reason to instead define `values.pinnipedProxy.URL` or similar as discussed on the previous PR. I'll do that in a follow-up so we can discuss the change there.